### DISCUSSION TITLE
Tolerate missing columns / values on INSERT from Pandas DataFrame

### DISCUSSION THREAD

**Discussion by davetapley** [duckdb/duckdb#7465] [Ideas]
I'd love it if duckdb could optionally tolerate missing columns / values on `INSERT` from Pandas DataFrame by filling `NULL`s.
Presently it throws e.g: `Binder Error: table iv has 5 columns but 4 values were supplied`.

It can fix by massaging the DataFrame first, but wanted to put down here in case there was interest from others.

I elaborate on why here:
- https://github.com/DOI-USGS/dataretrieval-python/issues/101

**Comment by Mytherin** [comment]
Have you seen the new `by_name` option in `append` (see #7300)? Perhaps that solves the issue

**Reply by davetapley** [reply]
Nice! It looks like exactly what I need, and I'm glad I tried it pre-release because:
- https://github.com/duckdb/duckdb/issues/7472

Happy accident that my dataset has numeric column names 😁

**Comment by Mytherin** [comment]
I have also pushed a PR to add support for a `BY NAME` option to the SQL `INSERT` (#7475)
